// opskins-express-trade/libs/classes/Trade.js
class Trade {
    constructor(request) {
        this.request = request
    }

    acceptOffer(params) {
        return this.request.post('ITrade/AcceptOffer/v1/', params)
    }

    cancelOffer(params) {
        return this.request.post('ITrade/CancelOffer/v1/', params)
    }

    getApps() {
        return this.request.get('ITrade/GetApps/v1/', {})
    }

    getOffer(params) {
        return this.request.get('ITrade/GetOffer/v1/', params)
    }

    getOffers(params) {
        return this.request.get('ITrade/GetOffers/v1/', params)
    }

    getTradeURL() {
        return this.request.get('ITrade/GetTradeURL/v1/', {})
    }

    getUserInventory(params) {
        return this.request.get('ITrade/GetUserInventory/v1/', params)
    }

    getUserInventoryFromSteamId(params) {
        return this.request.get('ITrade/GetUserInventoryFromSteamId/v1/', params)
    }

    regenerateTradeURL() {
        return this.request.post('ITrade/RegenerateTradeURL/v1/', {})
    }

    sendOffer(params) {
        return this.request.post('ITrade/SendOffer/v1/', params)
    }

    sendOfferToSteamId(params) {
        return this.request.post('ITrade/SendOfferToSteamId/v1/', params)
    }
}

module.exports = Trade
